core/src/main/java/org/mule/runtime/core/internal/event/EventQuickCopy.java

```diff
@@ -86,9 +85,7 @@ public static InternalEvent quickCopy(CoreEvent event, Map<String, Object> inter
     if (event instanceof EventQuickCopyInternalParametersDecorator) {
       final EventQuickCopyInternalParametersDecorator quickCopy = (EventQuickCopyInternalParametersDecorator) event;
 
-      final Map<String, Object> resolvedParams =
-          new HashMap<>(2 * (quickCopy.internalParameters.size() + internalParameters.size()));
-      resolvedParams.putAll(quickCopy.internalParameters);
+      final Map<String, Object> resolvedParams = (Map<String, Object>) copy(quickCopy.internalParameters);
```

Comment:
is the cast needed? doesn't the copy method declare the generic types already?

Reply:
For some reason, it won't compile otherwise 🤷‍♂